### ISSUE TITLE
More control over full static generated payloads cache policy

### ISSUE THREAD

**Issue by kedrzu** [nuxt/nuxt.js#7646]
### TLDR

I found this file: https://github.com/nuxt/nuxt.js/blob/917adc06184efd55a48123269b659adb288a3341/packages/config/src/config/generate.js
It seems that it's already possible to configure static payloads in the config file, but there is no mention about that in the docs.

### Is your feature request related to a problem? Please describe.

I'm diving into new cool feature of full static app generation.
I would like to have more control over, where those files go and how they should be named.

Especially when using with `@nuxt/pwa`, having all those files to sit in the same directory as other scripts means, that I have a limited possibility to control the cache policy. All scripts use a hash-based cache busting technique, and are forced to reload when name changes. That's why they sit in some folder, which name changes every build (based on timestamp).

This seems to be suboptimal, because for every build all the payloads will be invalided, even, if the payload did not change at all.

### Describe the solution you'd like

If it would be possible to completely alter naming policy for payload files and directory they sit in, it would be possible to use must-revalidate technique instead of pure cache-busting. Payload file would always have the same URL, but server can use ETag header to determine, whether browser needs to invalidate the cache or it can use cached version. Along with HTTP2, having additional round-trips to check caches is not a big problem.

This is a technique that is used by Netlify (https://www.netlify.com/blog/2017/02/23/better-living-through-caching/).
If it was possible to have additional control over naming, it would be also possible to make this kind of optimization.

The technique currently used have an advantage of always working, no matter the server configuration.
For more advanced usage, would be great to have a choice 😉.

**Comment by kedrzu** [comment]
After some experimenting I managed to place static payloads into `/static/` folder, using this config:

```
staticAssets: {
    base: '/static',
    versionBase: '/static',
    version: './',
    dir: '../static',
}
```
